feat: adding `max-retries` limit to preupgrade retry limit (#10137)
cosmos/cosmos-sdk: cosmovisor/README.md

> * `DAEMON_RESTART_AFTER_UPGRADE` (*optional*, default = `true`), if `true`, will restart the subprocess with the same command-line arguments and flags (but with the new binary) after a successful upgrade. Otherwise (`false`), `cosmovisor` stops running after an upgrade and requires the system administrator to manually restart it. Note that `cosmovisor` will not auto-restart the subprocess if there was an error.

Reviewer: ```suggestion
* `DAEMON_RESTART_AFTER_UPGRADE` (*optional*, default = `true`), if `true`, restarts the subprocess with the same command-line arguments and flags (but with the new binary) after a successful upgrade. Otherwise (`false`), `cosmovisor` stops running after an upgrade and requires the system administrator to manually restart it. Note restart is only after the upgrade and does not auto-restart the subprocess after an error occurs.
```

Reviewer: check this bit @spoo-bar to be sure I understand and explain well

Author: Addressed in #10209

> * `DAEMON_POLL_INTERVAL` is the interval length in milliseconds for polling the upgrade plan file. Default: 300.

Reviewer: tip: I have admired setting names to include unit of time, so a quick scan of the variable would require no fine print. In this case, I would suggest naming the variable
`DAEMON_POLL_INTERVAL_MS`

> * `UNSAFE_SKIP_BACKUP` (defaults to `false`), if set to `false`, will backup the data before trying the upgrade. Otherwise it will upgrade directly without doing any backup. This is useful (and recommended) in case of failures and when needed to rollback. It is advised to use backup option, i.e., `UNSAFE_SKIP_BACKUP=false`

Reviewer: ```suggestion
* `UNSAFE_SKIP_BACKUP` (defaults to `false`), if set to `false`, backs up the data before trying the upgrade. Otherwise (`true`), upgrades directly without performing a backup. The default value of false is useful and recommended in case of failures and when a backup needed to rollback. We recommend using the default backup option `UNSAFE_SKIP_BACKUP=false`.
```

Author: Addressed in #10209

> * `DAEMON_PREUPGRADE_MAX_RETRIES` (defaults to `0`) is the maximum number of times `pre-upgrade` is called in the application after exit status of `31`. More info on exit status codes [here](../docs/migrations/pre-upgrade.md). After the specified number of retires, if the pre-upgrade does not succeed, cosmovisor fails the upgrade. 

Reviewer: ```suggestion
* `DAEMON_PREUPGRADE_MAX_RETRIES` Default: 0. The maximum number of times to call `pre-upgrade` in the application after exit status of `31`. After the maximum number of retries, cosmovisor fails the upgrade. 
```

Reviewer: tip: instead of `[here]` it's useful to provide descriptive link text https://developers.google.com/style/link-text to inform our readers
